lib/settings/presentationSettings.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+var hooks = require('../hooks/hooks.js');
+
+
+var Promise     = require("bluebird")  
+  , coroutine   = Promise.coroutine
+  , fs = require("fs");
+
+var Exercise = db.model('Exercise');
+var Slideshow = db.model('Slideshow');
+var Setting = db.model('Setting');
+var ObjectId = require('mongoose').Types.ObjectId;
+
+Promise.promisifyAll(fs);
+
+
+module.exports = {
+
+    // TODO: need this?
+    getDustifySettingsOfExercisesAll: coroutine(function* getDustifySettingsOfExercisesAll(slideshowId) {
+    var slideshow = yield Slideshow.findById(slideshowId).exec();
+    var slides = slideshow.exercisesPerSlide;
+    var exerciseSettings = [];
+
+    for (var key in slides) {
+      if (slides.hasOwnProperty(key)) {
+        var singleSlide = {
+          index: key,
+          exercises: []
+        }
+        for ( var i=0; i<slides[key].length; i++ ) {
+          var exercise = yield Exercise.findById(slides[key][i]).exec();
```

Comment:
this will make a lot of requests in the db, one each for exercise. aggregate the exercise id's in an array, create one query with `Slideshow.find(_id : {$in: <your-array>}).exec()`. and then do a second loop to populate `exerciseSettings`